tuf/scripts/repo.py

```diff
@@ -282,24 +283,35 @@ def delegate(parsed_arguments):
       os.path.join(parsed_arguments.path, REPO_DIR))
 
   if parsed_arguments.role == 'targets':
-    repository.targets.delegate(parsed_arguments.delegatee, public_keys,
-        parsed_arguments.delegate, parsed_arguments.threshold,
-        parsed_arguments.terminating, list_of_targets=None,
-        path_hash_prefixes=None)
+    parent_targets = repository.targets
+    # Use key from --sign if present or use the default targets_key
+    parent_targets_keys = parsed_arguments.sign or [os.path.join(KEYSTORE_DIR, TARGETS_KEY_NAME)]
+  else:
+    parent_targets = repository.targets(parsed_arguments.role)
+    parent_targets_keys = parsed_arguments.sign
+
+  parent_targets.delegate(parsed_arguments.delegatee, public_keys,
+    parsed_arguments.delegate, parsed_arguments.threshold or 1,
+    parsed_arguments.terminating, list_of_targets=None,
+    path_hash_prefixes=None
+  )
 
+  # Load the private key for the parent targets
+  for key in parent_targets_keys:
     targets_private = import_privatekey_from_file(
-        os.path.join(parsed_arguments.path, KEYSTORE_DIR, TARGETS_KEY_NAME),
-        parsed_arguments.targets_pw)
+      os.path.join(parsed_arguments.path, key),
+      parsed_arguments.targets_pw
```

Comment:
Are all the keys expected to share the same password? That's a strange assumption...